### ISSUE TITLE
Regression: `sass` `compile*` methods fail in Bun v1.1.30 with `sourceMap: true`

### ISSUE THREAD

**Issue by jakeboone02** [oven-sh/bun#14414]
### What version of Bun is running?

1.1.30+7996d06b8

### What platform is your computer?

`Linux 5.15.153.1-microsoft-standard-WSL2 x86_64 x86_64`
and same thing happens on
`Microsoft Windows NT 10.0.22631.0 x64`

### What steps can reproduce the bug?

This is probably a more general problem not specific to Sass, but it's where I found it.

All `compile*` methods from the `sass` package fail when the `sourceMap` option is `true`.

> _Edit:_ here's a one-liner repro:
> ```bash
> bun -e 'import { compileString } from "sass"; compileString(".c{color: red}", { sourceMap: true });'
> ```

```ts
import { compileString } from "sass"; // sass@1.79.4

const scssString = `.ruleGroup {
  display: flex;
  flex-direction: column;
  gap: 0.5rem;
  padding: 0.5rem;
  border-width: 1px;
}
`;

// Works in all versions:
compileString(scssString, { sourceMap: false });

// Fails in v1.1.30, works in v1.1.29:
compileString(scssString, { sourceMap: true });
```

### What is the expected behavior?

Should not error.

### What do you see instead?

```
$ bun index.ts
{
  css: ".ruleGroup{display:flex;flex-direction:column;gap:.5rem;padding:.5rem;border-width:1px}",
  loadedUrls: [],
}
2296 |     },
2297 |     checkNum(value) {
2298 |       return value;
2299 |     },
2300 |     wrapException(ex) {
2301 |       return A.initializeExceptionWrapper(new Error(), ex);
                                                 ^
error: [BUG] Unreachable
      at wrapException (/home/jaboone/.bun/install/cache/sass@1.79.4@@@1/sass.dart.js:2301:43)
      at throwExpression (/home/jaboone/.bun/install/cache/sass@1.79.4@@@1/sass.dart.js:2320:15)
      at mapSpan$1 (/home/jaboone/.bun/install/cache/sass@1.79.4@@@1/sass.dart.js:107159:16)
      at call$0 (/home/jaboone/.bun/install/cache/sass@1.79.4@@@1/sass.dart.js:112779:52)
      at get$span (/home/jaboone/.bun/install/cache/sass@1.79.4@@@1/sass.dart.js:107618:88)
      at get$start (/home/jaboone/.bun/install/cache/sass@1.79.4@@@1/sass.dart.js:107656:21)
      at forSpan$1$2 (/home/jaboone/.bun/install/cache/sass@1.79.4@@@1/sass.dart.js:116823:50)
      at forSpan$2 (/home/jaboone/.bun/install/cache/sass@1.79.4@@@1/sass.dart.js:116832:19)
      at visitCssStyleRule$1 (/home/jaboone/.bun/install/cache/sass@1.79.4@@@1/sass.dart.js:114833:10)
      at accept$1$1 (/home/jaboone/.bun/install/cache/sass@1.79.4@@@1/sass.dart.js:117925:22)

Bun v1.1.30 (Linux x64)
```

### Additional information

Bun v1.1.29 did not have this issue.

**Comment by Jarred-Sumner** [comment]
This appears to be a JavaScriptCore regression - https://github.com/oven-sh/bun/compare/a01f9d8e1b39aeb2fba009dbc4b23b8760cdfd0d...a15244a4c925c049ef9097fcdae282ee898a0f8c

**Comment by jakeboone02** [comment]
@Jarred-Sumner for the test cases above, this appears to be fixed in 1.1.31.
